deps/amqp10_client/test/system_SUITE.erl

```diff
@@ -324,61 +347,107 @@ roundtrip_large_messages(Config) ->
+                                      durable => unsettled_state,
+                                      capabilities => SenderCapabilities}},
+                   snd_settle_mode => settled,
+                   rcv_settle_mode => first,
+                   filter => #{},
+                   properties => #{}
+                   },
+    {ok, Sender} = amqp10_client:attach_link(Session, SenderAttachArgs),
+    %%await_link(Sender, credited, link_credit_timeout),
+    await_link(Sender, attached, attached_timeout),
 
     Now = os:system_time(millisecond),
-    Props = #{creation_time => Now,
-              message_id => <<"my message ID">>,
-              correlation_id => <<"my correlation ID">>,
+    Props = #{content_encoding => <<"my content encoding">>,
               content_type => <<"my content type">>,
-              content_encoding => <<"my content encoding">>,
-              group_id => <<"my group ID">>},
+              correlation_id => <<"my correlation ID">>,
+              creation_time => Now,
+              group_id => <<"my group ID">>,
+              message_id => <<"my message ID">>,
+              to => <<"localhost">>
+              },
     Msg0 = amqp10_msg:new(<<"my-tag">>, Body, true),
     Msg1 = amqp10_msg:set_application_properties(#{"a_key" => "a_value"}, Msg0),
     Msg2 = amqp10_msg:set_properties(Props, Msg1),
-    Msg = amqp10_msg:set_message_annotations(#{<<"x-key">> => "x-value",
-                                               <<"x_key">> => "x_value"}, Msg2),
+    Msg = amqp10_msg:set_message_annotations(MessageAnnotations, Msg2),
     ok = amqp10_client:send_msg(Sender, Msg),
     ok = amqp10_client:detach_link(Sender),
     await_link(Sender, {detached, normal}, link_detach_timeout),
 
     {error, link_not_found} = amqp10_client:detach_link(Sender),
-    {ok, Receiver} = amqp10_client:attach_receiver_link(
-                       Session, <<"banana-receiver">>, <<"test1">>, settled, unsettled_state),
+    ReceiverAttachArgs = #{
+                   name => <<"banana-receiver">>,
+                   role => {receiver, #{address => Destination,
+                                        durable => unsettled_state,
+                                        capabilities => ReceiverCapabilities}, self()},
+                   snd_settle_mode => settled,
+                   rcv_settle_mode => first,
+                   filter => #{},
+                   properties => #{}
+                   },
+    {ok, Receiver} = amqp10_client:attach_link(Session, ReceiverAttachArgs),
     {ok, OutMsg} = amqp10_client:get_msg(Receiver, 4 * 60_000),
     ok = amqp10_client:end_session(Session),
     ok = amqp10_client:close_connection(Connection),
 
     % ct:pal(?LOW_IMPORTANCE, "roundtrip message Out: ~tp~nIn: ~tp~n", [OutMsg, Msg]),
-    ?assertMatch(Props, amqp10_msg:properties(OutMsg)),
+    ActualProps = amqp10_msg:properties(OutMsg),
+    [ ?assertEqual(V, maps:get(K, ActualProps)) || K := V <- Props, K =/= creation_time],
```

Comment:
The creation-time must not be changed by the broker as it's part of the properties which are part of the bare message and therefore immutable. If IBM MQ changes this creation time, this is a bug in IBM MQ. Can we maintain the assertion that creation time matches, and only omit this assertion for the buggy IBM MQ?